### PR TITLE
pallet_uniques : Impl set_collection_attribute and set_attribute

### PR DESCRIPTION
**Pull request by stanly-johnson**
Changelog:
- Extract set_attribute logic to `do_set_attribute` fn
- Add `set_collection_attribute` and `set_attribute` function to `non_fungibles::Mutate` implementation

### CODE REVIEW

File: frame/uniques/src/impl_nonfungibles.rs [hunk 1]
```diff
@@ -142,6 +142,31 @@ impl<T: Config<I>, I: 'static> Mutate<<T as SystemConfig>::AccountId> for Pallet
 			Ok(())
 		})
 	}
+
+	fn set_collection_attribute(
```
Comment (ggwpez) [inline frame/uniques/src/impl_nonfungibles.rs:146]: This function is unused?

**Comment by stanly-johnson** [comment]
> Hi @stanly-johnson 👋
> Could you please explain why this change is necessary and why you want to do it?

Hi, want to manage creation/management of Nonfungible tokens from another pallet. Expecting a config that implements `non_fungibles::Mutate` and using pallet-uniques but realised that uniques does not implement the functions that allow setting the collection/item attritbutes. Thought this is the better way rather than tightly coupling pallet-uniques.

File: frame/uniques/src/lib.rs [hunk 2]
```diff
@@ -375,6 +375,10 @@ pub mod pallet {
 		MaxSupplyAlreadySet,
 		/// The provided max supply is less to the amount of items a collection already has.
 		MaxSupplyTooSmall,
+		/// The provided key is out of bounds
```
Comment (ggwpez) [inline frame/uniques/src/lib.rs:378]:
```suggestion
		/// The provided key is too long.
```

File: frame/uniques/src/impl_nonfungibles.rs [hunk 3]
```diff
@@ -142,6 +142,31 @@ impl<T: Config<I>, I: 'static> Mutate<<T as SystemConfig>::AccountId> for Pallet
...
+		collection: &Self::CollectionId,
+		key: &[u8],
+		value: &[u8],
+	) -> DispatchResult {
+		let bounded_key: BoundedVec<u8, T::KeyLimit> =
+			key.clone().to_vec().try_into().map_err(|_| Error::<T, I>::KeyTooLong)?;
+		let bounded_value: BoundedVec<u8, T::ValueLimit> =
```
Comment (ggwpez) [inline frame/uniques/src/impl_nonfungibles.rs:153]: Could you please add some tests that these errors are triggered?  
Same for the `set_attribute`.

**Review by ggwpez — COMMENTED** [review]
Now I get it, yea makes sense to use the trait here.

File: frame/uniques/src/impl_nonfungibles.rs (hunk 3 above)
Comment (stanly-johnson) [reply frame/uniques/src/impl_nonfungibles.rs:153]: https://github.com/paritytech/substrate/pull/11544/commits/ec38c47d55d4a4db1ea8e838b39cbe5bd5f5935b

**Pull request closed without merging**